### ISSUE TITLE
False compilation failure with ranges

### ISSUE THREAD

**Issue by samwhitlock** [andreasfertig/cppinsights#410]
* [here is a link to the source in cppinsights](https://cppinsights.io/s/35c6b453)
* This should definitely compile (compiler explorer is totally fine with it)
* It seems to fail a concept check for `ranges::begin` on a vector, which is a bit perplexing

**Comment by samwhitlock** [comment]
Just for posterity's sake:

```cpp
#include <ranges>
#include <concepts>
#include <vector>

using namespace std;

int main()
{
  /*
  auto x = std::views::drop(3) | std::views::take(1);
  vector<int> vv{1,2,3,4};
  auto rng = vv | std::views::take(3);
  auto sz = std::ranges::size(rng);
  */
  std::vector vec{1, 2, 3, 4, 5, 6};
 auto v = vec | std::views::reverse | std::views::drop(2);
}
```

**Comment by andreasfertig** [comment]
Hello @samwhitlock,

thanks for reporting this. What you're looking at isn't an issue in C++ Insights. It is that Clang and with that C++ Insights, by default uses libstdc++ on Linux (GCC STL). Now, libstdc++ is written for GCC but also supports Clang. However, in transition periods like now, it can be that libstdc++ and Clang are not 100% compatible. I don't know whether it is something in Clang that is missing or something in libstdc++. If you compile the example with Clang in compiler explorer it doesn't compile [godbolt.org/z/3eGx8Wv5M](https://godbolt.org/z/3eGx8Wv5M). 

C++ Insights by default uses libstdc++ but has an option for using libc++ instead (where you select the desired standard). In the case of ranges switching the STL is no win as libc++ hasn't implemented ranges yet.

While likely not satisfying, I hope it helps.

Andreas

**Comment by samwhitlock** [comment]
Definitely! i hope clang gets this support soon, as some of this stuff about how views are constructed will be great to see in cppinsights, especially the "lazy pipe" stuff, e.g. `auto x = std::views::drop(3) | std::views::take(1);`. Thank you!

(you can close this issue if you wish, imo)
